iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/SV.java

```diff
@@ -6,67 +6,57 @@
  */
 package com.powsybl.iidm.network.util;
 
+import com.powsybl.commons.PowsyblException;
 import com.powsybl.iidm.network.DanglingLine;
 import com.powsybl.iidm.network.Line;
 import com.powsybl.iidm.network.TieLine;
 import com.powsybl.iidm.network.TwoWindingsTransformer;
+
 import org.apache.commons.math3.complex.Complex;
 import org.apache.commons.math3.complex.ComplexUtils;
 
 /**
  * Utility class to compute the state variables on one side of a branch, knowing
  * the state variables on the other side.
  *
+ *  | Y11   Y12 ||V1|   |I1|
+ *  |           ||  | = |  |
+ *  | Y21   Y22 ||V2|   |I2|
+ *
+ *  I1 = S1* / V1* and I2 = S2* / V2*
+ *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
  */
 public class SV {
 
-    public static double getRho(TwoWindingsTransformer twt) {
-        return twt.getRatedU2() / twt.getRatedU1() * twt.getOptionalRatioTapChanger().map(rtc -> rtc.getCurrentStep().getRho()).orElse(1d);
-    }
-
-    public static double getR(TwoWindingsTransformer twt) {
-        return twt.getR() * twt.getOptionalRatioTapChanger().map(rtc -> 1 + rtc.getCurrentStep().getR() / 100).orElse(1d);
-    }
-
-    public static double getX(TwoWindingsTransformer twt) {
-        return twt.getX() * twt.getOptionalRatioTapChanger().map(rtc -> 1 + rtc.getCurrentStep().getX() / 100).orElse(1d);
-    }
-
-    public static double getG(TwoWindingsTransformer twt) {
-        return twt.getG() * twt.getOptionalRatioTapChanger().map(rtc -> 1 + rtc.getCurrentStep().getG() / 100).orElse(1d);
+    public SV(double p, double q, double u, double a, int end) {
+        this.p = p;
+        this.q = q;
+        this.u = u;
+        this.a = a;
+        this.end = end;
     }
 
-    public static double getB(TwoWindingsTransformer twt) {
-        return twt.getB() * twt.getOptionalRatioTapChanger().map(rtc -> 1 + rtc.getCurrentStep().getB() / 100).orElse(1d);
+    /**
+     * @deprecated Not used anymore. The end associated to the voltage and flow must be defined. Use
+     * {@link SV#SV(double, double, double, double, int)}
```

Comment:
If we use `Branch.Side` type to specify the end instead of an `int` we would enforce the fact that this utility class is designed for calculating state variables at ends of **branches**.

Reply:
Done